comp/dogstatsd/statsd/statsd.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package statsd
+
+import (
+	"net"
+	"os"
+	"strconv"
+	"sync"
+
+	"go.uber.org/fx"
+
+	ddgostatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
```

Comment:
Please split the interface and implementation following those [guidelines](https://github.com/DataDog/datadog-agent/blob/main/docs/components/defining-components.md#defining-components).